aten/src/ATen/GeneratorForPrivareuseone.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <c10/util/intrusive_ptr.h>
+
+namespace at {
+
+using GeneratorFuncType = std::function<at::Generator(c10::DeviceIndex)>;
+
+static std::map<std::string, GeneratorFuncType>& GetGeneratorPrivateClassMap() {
+  static std::map<std::string, GeneratorFuncType> generator_privateuse1_map;
+  return generator_privateuse1_map;
+}
+
+class Register{
+public:
+  Register(std::string str, GeneratorFuncType func) {
+    TORCH_CHECK(GetGeneratorPrivateClassMap().size() == 0,
+      "Only can register the Generator for `privateuseone` once!");
+    GetGeneratorPrivateClassMap().insert(std::make_pair(str, func));
+  }
+};
+
+at::Generator GetGeneratorClassForPrivateuse1(c10::DeviceIndex device_index,
+    std::string str="privateuseone") {
```

Comment:
I think the string argument (and the first assert) is unnecessary: It's fine to just hardcode the assumption that PrivateUse1 == external backend registration key.